### ISSUE TITLE
Allow user to set a custom date period in the cost explorer widget

### ISSUE THREAD

**Issue by mlabouardy** [tailwarden/komiser#854]
**Describe the bug**
Currently, users can filter out the costs with the following date shortcuts:
- Last month
- Last 3 months
- Last 6 months
- Last 12 months
![Screenshot 2023-06-14 at 14 59 12](https://github.com/tailwarden/komiser/assets/10320205/2c35eae3-5969-428a-b15b-ca6f421b5c44)

**Additional context**
If you want to tackle this task, feel free to share with us your email address so we can give you access to the Figma project :)

**Comment by AlexVCS** [comment]
hi, I'd like to work on this issue, my email is alexcurtisslep@gmail.com

**Comment by jakepage91** [comment]
All yours @AlexVCS !

**Comment by AlexVCS** [comment]
Thanks! Can you give me access to the Figma project? I imagine there are mockups there :)

**Comment by AlexVCS** [comment]
I'm trying to run the dashboard locally. I think the go mod download command isn't working. I have version 1.19.3 of Go installed, I looked at the mod file and it uses 1.20, maybe that is the issue? When I run the command I don't see any error or progress, it just takes a few and then I can type again in my terminal.

I was able to cd into the dashboard, run npm i, but when I try to start the app (`npm run dev` from within the dashboard folder) it gives me this error:

`Error: Server could not be found. Refer to the logs and try again.

Source
components/layout/hooks/useTelemetry.tsx (14:14) @ eval

  12 | settingsService.getTelemetry().then(res => {
  13 |   if (res === Error) {
> 14 |     throw new Error(
     |          ^
  15 |       'Server could not be found. Refer to the logs and try again.'
  16 |     );
  17 |   } else {`

Any suggestions? Thanks!

**Comment by ShubhamPalriwala** [comment]
Hey @AlexVCS, Please take a look at this amazingly detailed video by Jake about contributing to the dashboard: https://www.youtube.com/watch?v=uwxj11-eRt8

This should solve the error you are getting! Let me know if it still persists, thanks!

**Comment by AlexVCS** [comment]
Hey @ShubhamPalriwala, thanks for sharing that video. I haven't created a config.toml or used cloud providers for the most part. I made a config.toml file in the root directory but am not all too sure where and how to get the credentials the documentation is saying I need. Is there more in depth information for someone who's never worked with this type of stuff before?

**Comment by ShubhamPalriwala** [comment]
Hey, thanks for reaching out! I totally understand your point and it can get overwhelming at the beginning, don't worry we'll get through it haha! So
- Install the Komiser CLI and start Komiser: Follow this video: https://docs.komiser.io/docs/introduction/getting-started#komiser-cli-installation
- Connect to a cloud provider of your choice: https://docs.komiser.io/docs/introduction/getting-started#self-hosted

Since your feature involves the cost explorer widget in the dashboard, connecting to any of the provider with atleast 1 resource should be enough!

If you get stuck anywhere or need any sort of help, hop on to our [Discord](https://discord.tailwarden.com/) and we can get you up and running 💯

**Comment by shobhitexe** [comment]
> Hey @ShubhamPalriwala, thanks for sharing that video. I haven't created a config.toml or used cloud providers for the most part


Hey, although I'm not entirely sure about the issue you're facing, based on my understanding, you can follow the steps below to establish a connection with your cloud provider (AWS in this case).

Goto Profile > Security credentials > scroll down to access keys > generate one 

![image](https://github.com/tailwarden/komiser/assets/29349136/7aea241b-37e6-407a-a5b0-0e335d169bdc)

Make a config.toml and credential file in root of project in my case i have named it cred.yaml

 like in below SS specify your credential file path

![image](https://github.com/tailwarden/komiser/assets/29349136/4aa7b5a3-08ec-4ff5-bb0d-4f7ea877c065)

in your credential file (cred.yaml in my case) put your key_id , access_key and region which you have obtained from aws in step 1

![image](https://github.com/tailwarden/komiser/assets/29349136/712be2f4-7113-4d07-b9db-0fcafec0d092)


I hope it helps

**Comment by AlexVCS** [comment]
thank you so much @shobhitexe, I really appreciate it! I have it running now :)

**Comment by ShubhamPalriwala** [comment]
thanks for the detailed help here @shobhitexe 🚀, @AlexVCS let us know if you need any help with this! let's get this going 🥇

**Comment by AlexVCS** [comment]
yeah a question: I brought in React-calendar as a package, is that ok or do you want me to integrate a calendar differently?

**Comment by ShubhamPalriwala** [comment]
hey @AlexVCS, it should be fine! its your feature, your opinion 💯 !

my personal approach to adding packages is the below checklist:
- [ ] can i do this without a package, if not
- [ ] find a minimal package with smallest dependencies and active contributions, if not
- [ ] least expected case, go with whatever works


see if the above checklist helps you and we should be gtg! i'm really excited about this feature 🚀

**Comment by karthick9019** [comment]
Hi team,
     We have added our AWS organization account in Komiser and configured via CLI. The thing is we are not getting the actual value which is present in our Billing console. Could you help out with this!!
